packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -148,26 +153,39 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
   }, [schemaFields, serverVersion]);
 
   const onFocus = () => {
-    if (hasAutofix && editorRef.current) {
-      if (editorRef.current.editorContents === '') {
+    if (hasAutofix && editorCurrentRef.current) {
+      if (editorCurrentRef.current.editorContents === '') {
         rafraf(() => {
-          editorRef.current?.applySnippet('\\{${}}');
+          editorCurrentRef.current?.applySnippet('\\{${}}');
         });
       }
     }
   };
 
   const onPaste = (event: React.ClipboardEvent<HTMLDivElement>) => {
-    if (hasAutofix && editorRef.current) {
-      const editorContents = editorRef.current.editorContents;
+    if (hasAutofix && editorCurrentRef.current) {
+      const editorContents = editorCurrentRef.current.editorContents;
       const snippet = lenientlyFixQuery(editorContents || '');
       if (snippet) {
-        editorRef.current.applySnippet(snippet);
+        editorCurrentRef.current.applySnippet(snippet);
         event.preventDefault();
       }
     }
   };
 
+  const onBlurEditor = useCallback(() => {
+    if (
+      !!editorCurrentRef.current &&
+      editorCurrentRef.current.editorContents !==
+        editorInitialValueRef.current &&
+      (editorInitialValueRef.current ||
+        editorCurrentRef.current.editorContents !== '{}')
```

Comment:
Is this checking for the auto-apply logic? As we only do this for components where `hasAutofix` is `true` then we probably want to account for that here too.

Also in cases where users got to this state on their own by removing input we will miss tracking. Not sure if it makes much sense to make this exception, but leave to you to decide

Reply:
I would keep the current `{}` check because of its simplicity compared to other options. We can revisit this decision if we lean toward tracking events even for autofix query edits.